### DISCUSSION TITLE
Are there any things to be considered before production?

### DISCUSSION THREAD

**Discussion by vigneshksaithal** [pocketbase/pocketbase#1528] [Q&A] [answered]
Like Firebase does Pocketbase has any limitations on document size? or any other things to be considered before production?

**Comment by ganigeorgiev** [comment, marked as answer]
The main recommendation before going to production is to setup a SMTP provider for sending emails. 
Other optional recommendation you could find in https://pocketbase.io/docs/going-to-production/#recommendations.

There is no hard limit for the collection record size. For general SQLite db limits you can check https://www.sqlite.org/limits.html.

**When upgrading a production deployed PocketBase app, make sure to always have a backup of your `pb_data` in case something goes wrong/unexpected.**

_Also please note that PocketBase is still under active development and full backward compatibility is not guaranteed before reaching [v1.0.0](https://github.com/orgs/pocketbase/projects/2)._

**Comment by vigneshksaithal** [comment]
I want to use Pocketbase for production, but I dont't know anything about server commands(like taking backup, adding ssl, etc). So, could there be a full tutorial of commands? OR could there be an official 1-Click app for Digitalocean, etc? I currently use Digitalocean marketplace app Pocketbase, but I don't know how to update it to 0.11. How can I proceed with it? Its a 1-Click app. I am finding difficult to update it.

**Reply by ganigeorgiev** [reply]
For a simple deployment setup (including auto ssl generation) you can follow the "Minimal setup" guide from https://pocketbase.io/docs/going-to-production/#minimal-setup (it is basically 3 commands + [optional copy-paste systemd service](https://pocketbase.io/docs/going-to-production/#systemd-service)).

Official 1-Click app for DO for now is not planned, because it will be hard to maintain. Once we reach a more stable version I'll consider creating one OR contribute to the existing one from https://github.com/pocketbase/pocketbase/discussions/1486#discussioncomment-4594748.
To update the above linked 1-Click install DO app, you can replace the binary located in [`/opt/pocketbase/pocketbase`](https://github.com/scott/pocketbase-do-packer/blob/abf54ba7381de46226a07b505a5843188047f0d1/scripts/01_setup_machine.sh#L20) via scp or rsync (_or ask the image maintainer to update it_). Here is an example with rsync:

```sh
rsync -avz -e ssh /local/path/to/pocketbase root@YOUR_SERVER_IP:/opt/pocketbase/pocketbase
```

---

The documentation can definitely be improved. I'll consider updating it in the near future when integrating the goja app hooks in v0.13/v0.14. But in general, some minimal familiarity with linux will be always needed. Suggestions how to simplify the self-hosting instructions are welcomed (_if you have an idea and time to contribute, the docs repo could be found in https://github.com/pocketbase/site_).

**Comment by vigneshksaithal** [comment]
@ganigeorgiev I am using DigitalOcean(DO) droplets to deploy PocketBase. As PocketBase stores all data in 1-file does it mean I need to scale it using higher storage VMs or can I scale it by adding DOs Volumes to it?

**Reply by ganigeorgiev** [reply]
If you want to increase only the storage size of your droplet, you can just mount an additional volume - https://www.digitalocean.com/products/block-storage.

As far as I know it acts as a "disk partition", so you'll need to place your `pb_data` within the volume mounting directory (you can explicitly specify the `pb_data` location with the `--dir` flag when starting PocketBase).
